Protocol.md

```diff
@@ -831,22 +874,33 @@ RESUME OK frames MUST always use Stream ID 0 as they pertain to the connection.
     |                         Stream ID = 0                         |
     +-----------+-+-+---------------+-------------------------------+
     |Frame Type |0|0|    Flags      |
+    +-------------------------------+---------------+---------------+
+    |0|               Number of Entries             |               
+    +-------------------------------+---------------+---------------+
+    |0|                         Stream ID                           |
     +-------------------------------+-------------------------------+
-    |0|                                                             |
-    +               Last Received Client Position                   +
-    |                                                               |
-    +---------------------------------------------------------------+
+    |                       Implied Position                        |
+    +-------------------------------+-------------------------------+
+    |0|                         Stream ID                           |
+    +-------------------------------+-------------------------------+
+    |                       Implied Position                        |
+    +-------------------------------+-------------------------------+
+                                   ...                               
 ```
 
 * __Frame Type__: (6 bits) 0x0E
-* __Last Received Client Position__: (63 bits = max value 2^63-1) Unsigned 63-bit long of the last implied position the server received from the client.
+* __Number of Entries__: (23 bits = max value 2,088,957) Unsigned 23-bit integer of the number of resumable streams being in progress.
+* __Stream ID__: (31 bits = max value 2^31-1 = 2,147,483,647) Unsigned 31-bit integer representing the resumable stream Identifier.
+* __Implied Position__: (32 bits = max value 2^32 = 4,294,967,296) Unsigned 32-bit integer of the individual Stream Resume Last Received Frame Position from the remote party. Value MUST be > 0.
 
 #### Keepalive Position Field
 
-Keepalive frames include the implied position of the client (or server). When sent, they act as a means for the other end of the connection to know the position of the other side.
+Keepalive frames include the set of implied positions of the client (or server). When sent, they act as a means for the other end of the connection to know the set of positions of the other side.
 
 This information MAY be used to update state for possible retransmission, such as trimming a retransmit buffer, or possible associations with individual stream status.
 
+Note, the positions of the stream which has not changed between Keepalives, must not be exchanged to keep Keepalive Frame size as small as possible.
```

Comment:
Is there a possibility that these get out of alignment? We previously sent total state in each keep alive.  Not we send effectively a set of deltas.

If they are out of alignment we potentially hide this from clients causing stalled upstream connections in a proxy for example.  How we we self repair in this case?

Should there be some way to checksum to detect they are out of alignment? Or possibility to provide some additional streams to verify the streams are still live on both sides?

Reply:
I guess that can be detected at the reconnection phase. However, we can use the global state for a sanity check and transfer it in addition to the set of individual streams. Having both values, the total number of emitted frames + the total number of received frames, and then keep alive sending individual patches, we can do the math and see if there is any diff

Reply:
Fixed that, now we send implied positions as before, so this should go out of aligment anymore